sdk/search/Azure.Search.Documents/src/Indexes/SearchableFieldAttribute.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Search.Documents.Indexes.Models;
+
+namespace Azure.Search.Documents.Indexes
+{
+    /// <summary>
+    /// Attributes a simple field using a primitive type or a collection of a primitive type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public class SearchableFieldAttribute : SimpleFieldAttribute, ISearchFieldAttribute
+    {
+        /// <summary>
+        /// Gets the data type of the field.
+        /// </summary>
+        /// <param name="collection">Whether the field is a collection of strings.</param>
+        public SearchableFieldAttribute(bool collection = false) : base(SearchFieldDataType.String, collection)
+        {
+        }
+
+        /// <summary>
+        /// Gets or sets the name of the language analyzer. This property cannot be set when either <see cref="SearchAnalyzerName"/> or <see cref="IndexAnalyzerName"/> are set.
+        /// Once the analyzer is chosen, it cannot be changed for the field in the index.
+        /// </summary>
+        public LexicalAnalyzerName? AnalyzerName { get; set; }
```

Comment:
Can we use our extensible enums as attribute values?  https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/attributes#attribute-parameter-types